### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by DzXiaoLMCD**
Fix or improve the issues list:
DEV-41301 Upgrade the go sdk to v2
DEV-49251 Improve panic log process
DEV-50734 Discover nodeport and loadbalance type services
DEV-50494 Add argus related identity for collector and collector group

### CODE REVIEW

**Comment by woz5999** [comment]
@DzXiaoLMCD this PR is failing the automated builds. It must pass to be merged

File: pkg/controller/collectorset.go [hunk 1]
```diff
@@ -38,6 +41,9 @@ func CreateOrUpdateCollectorSet(collectorset *crv1alpha1.CollectorSet, lmClient
 	}
 
 	secretIsOptional := false
+	// DEV-50734 convert to lower
```
Comment (woz5999) [inline pkg/controller/collectorset.go]: don't include internal tickets in public code

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -89,6 +91,8 @@ func (c *Controller) watch(ctx context.Context) error {
 }
 
 func (c *Controller) addFunc(obj interface{}) {
+	// Due to panic error in this call stack will crash the application; recovering those panics here could make our application robust.
+	defer err.RecoverError("Create collectorset")
```
Comment (woz5999) [inline pkg/controller/controller.go]: just fix the panic instead...

File: pkg/controller/controller.go [hunk 3]
```diff
@@ -143,6 +149,8 @@ func (c *Controller) updateFunc(oldObj, newObj interface{}) {
 }
 
 func (c *Controller) deleteFunc(obj interface{}) {
+	// Due to panic error in this call stack will crash the application; recovering those panics here could make our application robust.
```
Comment (woz5999) [inline pkg/controller/controller.go]: fix the panic

File: pkg/controller/controller.go [hunk 4]
```diff
@@ -123,6 +127,8 @@ func (c *Controller) addFunc(obj interface{}) {
 // TODO: updating the collectorset ids in the add func will trigger this. We
 // need to check for this case
 func (c *Controller) updateFunc(oldObj, newObj interface{}) {
+	// Due to panic error in this call stack will crash the application; recovering those panics here could make our application robust.
```
Comment (woz5999) [inline pkg/controller/controller.go]: fix the panic

File: pkg/err/err.go [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package err
```
Comment (woz5999) [inline pkg/err/err.go]: this is a really bad package name. "err" is an extremely common, almost idiomatic variable name. this package should be renamed.

File: pkg/controller/collectorset.go (hunk 1 above)
Comment (DzXiaoLMCD) [reply pkg/controller/collectorset.go]: OK, I will delete it

File: pkg/controller/controller.go (hunk 2 above)
Comment (DzXiaoLMCD) [reply pkg/controller/controller.go]: OK

File: pkg/controller/controller.go (hunk 3 above)
Comment (DzXiaoLMCD) [reply pkg/controller/controller.go]: OK

File: pkg/controller/controller.go (hunk 4 above)
Comment (DzXiaoLMCD) [reply pkg/controller/controller.go]: OK

File: pkg/err/err.go (hunk 5 above)
Comment (DzXiaoLMCD) [reply pkg/err/err.go]: OK, I will remove the err package and don't use it

**Comment by woz5999** [comment]
@DzXiaoLMCD Thanks for making the requested changes. I'm still curious about the reason RecoverError was implemented. We were previously seeing issues in those routines? I don't think that just catching every panic is the correct solution, but I want to make sure we're still addressing the underlying issue that required those changes in the first place. Maybe you can explain more.

**Pull request merged**